search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+# Discovery Search Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Search Endpoint
+**/api/discover/search**   
+
+This endpoint provides the functionality of the Discovery search screen: http://demo.dspace.org/xmlui/discover.
+
+Provide access to the Discovery search system (SOLR based). This endpoint returns the list of available DSpace object (DSO) types that can be searched:
+* all: Execute a query over all available DSO types.
+* items: Only search the DSpace items.
+* communities: Search within the DSpace community records.
+* collections: Limit the search to DSpace collection records.
+
+Example: TODO
+
+### Search with a given DSO type
+**/api/discover/search/<:dso-type>**
+
+This endpoint will provide detailed information on the Discovery search configuration that can be used to build complex searches.
+
+It supports the following parameters:
+* `scope`: UUID of a specific DSpace container (site, community or collection) to which the search will to be limited. If the scope has a specific DSpace configuration defined in `config/spring/api/discovery.xml`[https://github.com/DSpace/DSpace/blob/master/dspace/config/spring/api/discovery.xml#L28] that configuration will be returned. Otherwise the default configuration will be returned.
+
+The JSON response document is as follow
+```json
+{
+  "filters": [
+    {
+      "filter" : "title",
```

Comment:
Currently there is no way in the `discovery.xml` file to specify the allowed operators per filter. In DSpace 6 the set of operators is fixed (e.g. in [xmlui](http://demo.dspace.org/xmlui/discover?filtertype_1=title&filter_relational_operator_1=authority&filter_1=id%3F&submit_apply_filter=&query=) and [jspui](http://demo.dspace.org/jspui/simple-search?location=&query=&filtername=title&filtertype=authority&filterquery=id%3F&rpp=10&sort_by=score&order=desc)) 

While I agree that not having an "ID" option for filters without authority control is more userfriendly, this would require bigger changes to the dspace-api Discovery classes. This would make backporting DSpace 6 fixes harder. And since this is more a "nice to have" I would postpone it to a later time when we have time left.

Do you (and others) agree?